Sources/apollo-ios-pagination/CursorBasedPagination/CursorBasedPagination.swift

```diff
@@ -0,0 +1 @@
+public enum CursorBasedPagination { }
```

Comment:
Just used to namespace various cursor-based pagination strategies. I can think of multiple strategies -- with the two basic ones being implemented. Another strategy would be bi-directional, but the pager isn't currently built to support that. I'd argue that if bi-directional paging is needed, that it's a good case for a subclass _or_ a new class altogether.

Reply:
Is there really much value in having both `ForwardPagination` and `ReversePagination`? They functional exactly the same, it's just the names of the properties that are different, right? (eg. `hasNext` vs `hasPrevious`)

If you are doing reverse pagination, couldn't you just use the same `PaginationInfo`? I guess it makes the reasoning about "am I going backwards or forwards" more contextual knowledge the user has to reason about locally... But I feel like it's over kill to include them both in the library. If you _really_ want to do that, you can create your own `PaginationInfo` very easily.

Reply:
They're totally the same minus having different naming and contextual knowledge. In practice, I found it super confusing to instantiate a `CursorPage` (or `ForwardPage` or what have you) and then pass in the value for `hasPrevious` into `hasNext` and `startCursor` into `endCursor`). It looked like a mistake when looking at a diff.